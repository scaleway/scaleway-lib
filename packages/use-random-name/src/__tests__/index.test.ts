import { renderHook } from '@testing-library/react'
import { afterAll, beforeAll, describe, expect, it, vi } from 'vitest'
import useRandomName from '..'

describe('useRandomName', () => {
  beforeAll(() => {
    vi.spyOn(global.Math, 'random').mockReturnValue(0.4155913669444804)
  })

  afterAll(() => {
    vi.spyOn(global.Math, 'random').mockRestore()
  })

  it('useTranslation should not be defined without I18nProvider', () => {
    const { result } = renderHook(() => useRandomName())
    expect(result.current).toMatchSnapshot()
  })
})
