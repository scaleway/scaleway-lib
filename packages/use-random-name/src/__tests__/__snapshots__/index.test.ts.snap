// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useRandomName > useTranslation should not be defined without I18nProvider 1`] = `"intelligent-elgamal"`;
